viewer/src/components/AdvancedLog.vue

```diff
@@ -60,30 +63,36 @@ import { LogEntry } from "@gaia-project/engine";
         advancedLogIndex += 1;
         nextLogEntry = this.data.advancedLog[advancedLogIndex];
       };
+      const replaceMove = (s: string) => {
+        for (const e of replace) {
+          s = s.replace(e.from, e.to);
+        }
```

Comment:
In a 4p game, there will be 9 tech titles, 7 boosters, and 6 adv tech tiles, so 22 items in total.

On a 300 moves game, that mean 6,6k replaces.

You can get away with less.

First solution:

```ts
s = s.replace(/\btech-([a-z]+)/g, match => baseTechTileNames[data.tiles.techs[match].tile].name)
s = s.replace(/\badv-([a-z]+)/g, match => advancedTechTileNames[data.tiles.techs[match].tile])
// same with booster
```

That's three replaces only, instead of 22. And no need to precompute the replacement map with `createReplace`.

You could do only one:

```ts
s = s.replace(/\b((adv|tech)-([a-z]+)|booster[0-9]+)/g, match => {
  if (match.startsWith('booster')) {
    return ...
  } else if (match.startWith('adv') {
    
  } else {
  }
})
```

But maybe that's going too far.

I know there are probably other bottlenecks, like maybe we should not render advanced log on mobile or do it lazily with `v-intersect`: https://vuetifyjs.com/en/directives/intersect/#usage , but every little bit helps :)

Reply:
makes sense - wasn't more complicated than my original version.